github/src/main/kotlin/gropius/sync/github/Outgoing.kt

```diff
@@ -184,6 +186,82 @@ class Outgoing(
         }
     }
 
+    /**
+     * Mutate an AddedLabelEvent upto GitHub
+     * @param imsProjectConfig active config
+     * @param issueInfo info of the issue containing the timeline item
+     * @param userList users that have contributed to the event
+     * @return List of functions that contain the actual mutation executors
+     */
+    private suspend fun githubAddLabel(
+        imsProjectConfig: IMSProjectConfig, issueInfo: IssueInfo, label: Label, userList: Iterable<User>
+    ): List<suspend () -> Unit> {
+        logger.info("Scheduling adding ${label.name} (${label.rawId}) to ${issueInfo.neo4jId}")
+        val labelId = labelInfoRepository.findByNeo4jId(label.rawId!!)
+        if (labelId != null) {
+            return if (labelId.url == imsProjectConfig.url) {
+                listOf {
+                    val client = createClient(imsProjectConfig, userList)
+                    val response =
+                        client.mutation(MutateAddLabelMutation(issueInfo.githubId, labelId.githubId)).execute()
+                    val item =
+                        response.data?.addLabelsToLabelable?.labelable?.asIssue()?.timelineItems?.nodes?.lastOrNull()
+                    if (item != null) {
+                        incoming.handleTimelineEvent(imsProjectConfig, issueInfo, item)
+                    }
+                }
+            } else {
+                listOf();
+            }
+        } else {
+            return listOf {
+                val client = createClient(imsProjectConfig, userList)
+                val createLabelResponse = client.mutation(
+                    MutateCreateLabelMutation(
+                        issueInfo.githubId, label.name, label.description, label.color
+                    )
+                ).execute()
+                val newLabel = createLabelResponse?.data?.createLabel?.label?.labelData()
+                if (newLabel != null) {
+                    nodeSourcerer.ensureLabel(imsProjectConfig, newLabel)
+                    val response =
+                        client.mutation(MutateAddLabelMutation(issueInfo.githubId, newLabel.id)).execute()
+                    val item =
+                        response.data?.addLabelsToLabelable?.labelable?.asIssue()?.timelineItems?.nodes?.lastOrNull()
+                    if (item != null) {
+                        incoming.handleTimelineEvent(imsProjectConfig, issueInfo, item)
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Mutate an RemovedLabelEvent upto GitHub
+     * @param imsProjectConfig active config
+     * @param issueInfo info of the issue containing the timeline item
+     * @param userList users that have contributed to the event
```

Comment:
missing doc for label

Reply:
f4cea4b1d0191373ae90c82a78fcdbe78e5cf47d